qa/vector/src/main/java/org/elasticsearch/test/knn/KnnIndexTester.java

```diff
@@ -35,19 +39,26 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
+import java.util.Map;
 
 /**
  * A utility class to create and test KNN indices using Lucene.
  * It supports various index types (HNSW, FLAT, IVF) and configurations.
  */
 public class KnnIndexTester {
-    static final Level LOG_LEVEL = Level.DEBUG;
-
-    static final SysOutLogger logger = new SysOutLogger();
+    static final Logger logger;
```

Comment:
I like this change, however, it requires the formatting when using this logger to be changed. 

For example I see this log line written:

```
15:11:58.112 [main] INFO  org.elasticsearch.test.knn.KnnIndexTester - docsPathSizeInBytes=%d, dim=%d, vectorEncoding=%s, byteSize=%d
```

And
```
15:13:05.276 [main] INFO  org.elasticsearch.test.knn.KnnIndexTester - completed %d searches in %d ms: %d QPS CPU time=%dms
```
:D